### ISSUE TITLE
Not getting correct pageState

### ISSUE THREAD

**Issue by aamirafridi** [tes/seguir#59]
You will get pageState from sequir when you have 0 or 1 more feed to show.
Example:

**PageSize is set to 2 for testing/explaining the issue**

I have 2 feeds on my home page
![image](https://cloud.githubusercontent.com/assets/55896/11003276/607053da-84a9-11e5-9098-a993e11fbc70.png)

It returns pageState for next page.
I copy the page state and try to fetch the next page
![image](https://cloud.githubusercontent.com/assets/55896/11003298/8022eda0-84a9-11e5-9de3-9831ccebc155.png)

Now I will add 3rd feed locally localhost:5408/social-events/tes/post and try to fetch the feeds again and it gives me 2 feeds and next pageState which is fine/expected
When I take this pageState and try to access next page I expect 1 feed and no pageState and I get this correctly this time
![image](https://cloud.githubusercontent.com/assets/55896/11003412/3d8d6a46-84aa-11e5-8201-0ebd4836ff12.png)

I add another feed locally and I expect 2 feeds on I request for the 2nd page but here I get pageState again (not expected)
![image](https://cloud.githubusercontent.com/assets/55896/11003447/7ecb9474-84aa-11e5-8bfb-5ac8d50a1239.png)
Not sure what is the issue.

**Comment by aamirafridi** [comment]
cc @tomruttle / @dmorgantini / @cliftonc

**Comment by cliftonc** [comment]
Pagestate isn't what you think it is, we need to read the Cassandra docs.  

It is a way to page through a static response - not sure how Cassandra stores this state internally.

If you add new posts I would refresh the whole thing and go back to top (simulate the notification at the top being there and it reloading the feed with new content when clicked).

**Comment by aamirafridi** [comment]
this
![image](https://cloud.githubusercontent.com/assets/55896/11042328/53a917ce-870e-11e5-8c7f-3784fbed90d9.png)
 (feeds test in sequir)

**Comment by dmorgantini** [comment]
undefined means there are no more pages to load.

**Comment by aamirafridi** [comment]
Yes but when will it give me page state in 2nd run?
page state is the state of next page according to http://docs.datastax.com/en/developer/nodejs-driver/1.0/nodejs-driver/reference/paging.html

**Comment by dmorgantini** [comment]
the last character is different.

**Comment by dmorgantini** [comment]
sorry let me actually read the issue then I'll get back to you.

**Comment by dmorgantini** [comment]
Right - so the reason is that Cassandra only knows that there is another page but not how many in the next item.  As you can see in image 2 above even though there is nothing more to show you get a page state.  So therefore when the state of the feed changes cassandra is going to give you another page state if the current page becomes full because it doesn't know whether or not you added 1 or more it just knows that more have been added.

TL;DR: this shouldn't be an issue when there are more than 2 per page.

**Comment by aamirafridi** [comment]
The problem is that we have "LOAD MORE" button will be displayed when we have nextPageState and when clicked, we make an API request to fetch more feeds. So in this case, it when display the button which when clicked returns no feed so on UI the button will disappear since api will return no page state. I guess we remove the button and automatically load more feeds when user hits the bottom of page i.e auto load instead of load on user's action. That will solve the problem but its expensive as user might  not want to see more feeds?

Regards,
Aamir

**Comment by dmorgantini** [comment]
I don't think it's a problem.  It's not unheard of in other systems where you click load more - nothing more is displayed and the button disappears.  It's a reality of systems where you don't know how many pages you are going to get when you start paginating. We could simply add a box saying 'end of feed try following other users for more!' Or something more user friendly.

**Comment by dmorgantini** [comment]
The followers page has the same issues.

**Comment by dmorgantini** [comment]
let's chat tomorrow morning - I don't think I'm explaining very well.

**Comment by aamirafridi** [comment]
It make sense. I will close this for now if thats the case.
